### ISSUE TITLE
ci: browser testing is failing sometimes

### ISSUE THREAD

**Issue by aladdin-add** [eslint/eslint#18548]
### ESLint version

main

### What problem do you want to solve?

https://github.com/eslint/eslint/commit/2c8fd34bf1471efbd6e616b50d4e25ea858a6989 did not fully fix the browser testing - it's still sometimes failing - e.g. https://github.com/eslint/eslint/actions/runs/9353577317/job/25744423567

```bash
========= Your concise report ==========
chrome-headless-shell (v125.0.6422.112) on linux
❌ Failed to setup tests, no tests found
```

### Participation

- [ ] I am willing to submit a pull request for this change.

**Comment by fasttime** [comment]
Just tried running Karma and it still works. I don't recall it giving any headaches.

**Comment by aladdin-add** [comment]
I'm not surprised - Karma is very stable and has very few updates. I'm fine to switching back as we don't really need any advanced features, karma is already sufficient :)

**Comment by mdjermanovic** [comment]
Karma was working great for our simple use case, but is now [deprecated](https://github.com/karma-runner/karma?tab=readme-ov-file#karma) so we'd likely need to switch from Karma sooner or later anyway.

**Comment by aladdin-add** [comment]
it's happening again: https://github.com/eslint/eslint/actions/runs/10260943448/job/28387817330?pr=18754

@christian-bromann can you take a look, thanks! 🙏

**Comment by christian-bromann** [comment]
@aladdin-add looking into it.

**Comment by christian-bromann** [comment]
@aladdin-add I raised a PR, mind taking a look?

**Comment by christian-bromann** [comment]
Please keep me posted if you continue to see unreliable test runs. Thank you!

**Comment by aladdin-add** [comment]
@christian-bromann  In rare cases, it can still be happening:

```bash
Execution of 1 workers started at 2024-08-09T04:44:40.372Z

[0-0] RUNNING in chrome - file:///tests/lib/linter/linter.js
[0-0]  Error:  Test failed due to following error(s):
  - ignore.js?v=a53bdd1a: Failed to load resource: the server responded with a status of 504 (Outdated Optimize Dep)

  - escape-string-regexp.js?v=fee7ba7f: Failed to load resource: the server responded with a status of [50](https://github.com/eslint/eslint/actions/runs/10313769226/job/28551070874#step:5:51)4 (Outdated Optimize Dep)

  - natural-compare.js?v=a894a6e8: Failed to load resource: the server responded with a status of 504 (Outdated Optimize Dep)

[0-0] FAILED in chrome - file:///tests/lib/linter/linter.js
```

should have been fixed in https://github.com/webdriverio/webdriverio/pull/13331?

**Comment by christian-bromann** [comment]
Not sure, however we are releasing v9 of WebdriverIO next week which comes with some performance improvements for the component testing part which I hope will resolve this problem and make pipelines more stable.
